alethzero/plugins/whisper/Whisper.cpp

```diff
@@ -169,13 +170,16 @@ void Whisper::on_newIdentity_triggered()
 
 void Whisper::on_post_clicked()
 {
-	return;
 	shh::Message m;
 	m.setTo(stringToPublic(m_ui->shhTo->currentText()));
-	m.setPayload(parseData(m_ui->shhData->toPlainText().toStdString()));
+	m.setPayload(bytesConstRef(m_ui->shhData->toPlainText().toStdString()).toBytes());
+
 	Public f = stringToPublic(m_ui->shhFrom->currentText());
 	Secret from;
 	if (zero()->web3Server()->ids().count(f))
 		from = zero()->web3Server()->ids().at(f);
-	web3()->whisper()->inject(m.seal(from, topicFromText(m_ui->shhTopic->toPlainText()), m_ui->shhTtl->value(), m_ui->shhWork->value()));
+
+	shh::BuildTopic topic(m_ui->shhTopic->toPlainText().toStdString().c_str());
```

Comment:
Is .c_str() redundant?  Looks like it should accept a string (and c string is converted to a string).